lesson_22/orm/homework_22.py

```diff
@@ -0,0 +1,275 @@
+студентів, зареєстрованих на певний курс, або курси,
+на які зареєстрований певний студент.
+Оновлення та видалення даних: Реалізуйте
+можливість оновлення даних про студентів або курси,
+а також видалення студентів з бази даних.
+Можна використовувати будь яку ORM на Ваш вібир
+"""
+
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def populate_db_from_csv(db, courses_file, students_file):
+    """Populate the database with data from CSV files."""
+    try:
+        with open(courses_file, 'r') as course_file:
+            reader = csv.DictReader(course_file)
+            if 'name' not in reader.fieldnames:
+                raise KeyError("Missing 'name' column in courses file.")
+            courses = [db_model.Courses(title=row['name']) for row in reader]
+            db.session.add_all(courses)
+
+        with open(students_file, 'r') as stud_file:
+            read = csv.DictReader(stud_file)
+            if 'name' not in read.fieldnames or 'age' not in read.fieldnames:
+                raise KeyError("Missing 'name' or 'age' column in the file.")
+            students = [
+                db_model.Students(
+                    name=row['name'],
+                    age=row['age'],
+                ) for row in read
+            ]
+            db.session.add_all(students)
+
+        db.session.commit()
+        logging.info('Database populated successfully from CSV files!')
+    except (KeyError, FileNotFoundError) as err:
+        logging.error(f'Error with CSV file: {err}')
+    except SQLAlchemyError as err:
+        logging.error(f'Error populating database: {err}')
+
+
+def random_enrollment_assignment(db):
+    """Random enrollment assignments."""
+    try:
+        students = db.session.query(db_model.Students).all()
+        courses = db.session.query(db_model.Courses).all()
+
+        if not students or not courses:
+            logging.warning('No students or courses available for enrollment.')
+            return
+
+        unique_enrollments = set()
+        enrollments = []
+        # 3 is a random value that is used for 3 cycles of enrollments
+        for _ in range(len(students) * 3):
+            student = random.choice(students)
+            course = random.choice(courses)
+            key = (student.student_id, course.course_id)
+
+            if key not in unique_enrollments:
+                unique_enrollments.add(key)
+                enrollments.append(db_model.
+                                   Enrollments(student_id=student.student_id,
+                                               course_id=course.course_id))
+
+        db.session.add_all(enrollments)
+        db.session.commit()
+        logging.info('Random enrollment assignments completed successfully!')
+    except SQLAlchemyError as err:
+        db.session.rollback()
+        logging.error(f'Error : {err}')
+
+    finally:
+        db.session.close()
```

Comment:
In class BTW this operation could be placed into` __del__ `magic call. If organize all this into Testing class - then fixtures will be very helpful